src/test/java/com/wikia/webdriver/common/core/api/ApiCall.java

```diff
@@ -0,0 +1,85 @@
+package com.wikia.webdriver.common.core.api;
+
+import com.wikia.webdriver.common.core.Helios;
+import com.wikia.webdriver.common.core.annotations.User;
+import com.wikia.webdriver.common.logging.PageObjectLogging;
+
+import org.apache.http.client.ClientProtocolException;
+import org.apache.http.client.entity.UrlEncodedFormEntity;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.message.BasicNameValuePair;
+import org.openqa.selenium.WebDriverException;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+
+/**
+ * Created by wikia on 2015-08-27.
+ */
+public abstract class ApiCall {
+
+  private static String ERROR_MESSAGE = "Problem with API call";
+
+  protected static String URL_STRING = null;
+
+  protected ApiCall() {
+
+  }
+
+
+  abstract protected String getURL();
+
+  /**
+   * Return null if API call doesn't require to be logged in as specific user
+   *
+   * @return User to be logged in while executing API call
+   */
+  abstract protected User getUser();
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Reorder the modifiers to comply with the Java Language Specification. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AModifiersOrderCheck)